### DISCUSSION TITLE
Possibility of a policy against useless lifecycle scripts

### DISCUSSION THREAD

**Discussion by xsrvmy** [npm/feedback#696] [Registry] [answered]
A few packages added lifecycle scripts recently whose sole purpose is to print out protests against Russia to the console. Some have reported this breaking CI.

Is there a need for a policy to ban such lifecycle scripts which serve no technical purpose for the package?

Two possible edge cases:
1. Developer tries to get around this by creating package A depending on package B, where package B's sole purpose is to display the message, so that package B doesn't violate the policy.
2. Package A is licensed under the GPL, and uses a reasonable lifecycle script, that in addition prints the GPL preamble. A fork of A finds the lifecycle script unnecessary, but the GPL license may still require printing out the license, so the fork ends up with a lifecycle script that only prints out a message.

EDIT: I should note that this is somewhat of an extension to the "no console ads" rule.

**Comment by markcellus** [comment]
Interesting. If this were a policy, would this ultimately hurt consumers of banned packages? Would banning a package disable any future installs? How would a banned package work for consumers who have already installed it before the ban?

**Reply by xsrvmy** [reply]
I was just pitching the idea that there needs to be clearer messaging on whether this sort of scripts should be tolerated, because the response is ranging from basically nothing in one repo, to being spammed with issue saying malware in another.

And it's not an instant ban or anything. I was just wondering if NPM should do something because this happened with ads once. This kind of thing becomes a slippery slope. What if someone puts a more controversial message in the postinstall that is against my beliefs? (If I override the dependency, do users of my package also get the overriden dependency?)

**Comment by MylesBorins** [comment, marked as answer]
As npm we try not to be too prescriptive about what people publish to the registry. Our [open source terms](https://docs.npmjs.com/policies/open-source-terms) outline what is consider acceptable to publish.

If you feel that a package is being malicious and wish for us to remove it, please report the package as malware via the link on the packages page. If you feel that it is violating our policies in other ways that are not clearly malicious, e.g. malware, please feel free to open a ticket with support.

A decision to remove a package or block a package for it's use of scripts is not something we do lightly and requires extensive internal conversation to make a decision. I don't believe this is something that could be clearly documented via policy or managed via automation

**Reply by xsrvmy** [reply]
Yeah I guess ads are different in some sense because they exploit the npm repo commercially.
After posting this I realized that overrides are already a thing, so perhaps a special override for disabling a specific script is more appropriate.
